src/transformers/training_args.py

```diff
@@ -556,6 +556,11 @@ def _setup_devices(self) -> "torch.device":
             # python -m torch.distributed.launch --nproc_per_node=2 ./program.py
             from .integrations import is_deepspeed_available
 
+            # workaround for setups like notebooks where the launcher can't be used,
+            # but deepspeed requires a dist env.
+            if self.local_rank == -1:
+                self.local_rank = 0
```

Comment:
Could we at least check the env variable `LOCAL_RANK` exists and is 0 before doing this (which will be the case if the user follow the instructions)? Always changing this makes me a bit nervous.

Reply:
Yes, of course! Done.